### DISCUSSION TITLE
x_advance data exception

### DISCUSSION THREAD

**Discussion by CoKoHo** [harfbuzz/harfbuzz#4646] [General]
I'm new to harfbuzz,After using hb_shape_full on the Android platform, the x_advance exception of glyph_positions is obtained. Sometimes it is a negative number and sometimes it is a large positive number.

The following is the x_advance obtained by android and windows

**Comment by CoKoHo** [comment]
android x_advance is ：
-2097152
-908928
-1856832
-2097152
103351971
28023818
-1966080
-1326336

**Comment by behdad** [comment]
Why do you mean by "x_advance exception of glyph_positions"??

**Reply by CoKoHo** [reply]
Because I print the glyph_position obtained from hb_buffer_get_glyph_positions, x_advance is not normal

**Comment by CoKoHo** [comment]
windows x_advance is :
1408
576
1088
1408
1058
1235
1216
832
